### DISCUSSION TITLE
Qustion Regarding multiple Child Fetchers

### DISCUSSION THREAD

**Discussion by manu-mnnit** [Netflix/dgs-framework#298] [Q&A]
Hi @paulbakker ,

I have created multiple child fetchers in my project. It looks they run sequentially, is there any way to run them in parallel ?

E.g. 
schema:
type Query{
    a: [Entity]
}
type Entity{
    field1: Int
    field2: [b]
   field3: [c]
}

type b{
  f1: Int
}
type c{
 f2: String
}

@dgsData(ParentType="Query", field="a")
List<a> func1(){
         return [a];
}

@dgsData(ParentType="Entity", field="b")
List<b> func2(){
      return [b];
}
@dgsData(ParentType="Entity", field="c")
List<c> funn3(){
      return [c];
}

**Comment by Nokinori** [comment]
Hi @manu-mnnit,
I think, problem is that your code was written in blocking style. As much as I can say according code example in the question)
So the simplest solution is wrapping your code in CompletableFuture. 

Like this:
```java
@DgsData(ParentType="Query", field="a")
CompletableFuture<List<Entity>> func1()  {
    return CompletableFuture.supplyAsync(() -> Entity());
}

@DgsData(ParentType="Entity", field="b")
CompletableFuture<List<String>> func2() {
    return CompletableFuture.supplyAsync(() -> "b")
}
```

Check this docs for more info:
- https://netflix.github.io/dgs/datafetching/
- https://netflix.github.io/dgs/data-loaders/
